### PR TITLE
Check severity att while validating

### PR DESCRIPTION
**Pull request by dogancanbakir**
## Proposed changes
Closes [https://github.com/projectdiscovery/nuclei/issues/3528](https://github.com/projectdiscovery/nuclei/issues/3528)

## Checklist

- [X] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ShubhamRasal** [comment]
```
❯ ./nuclei -t templates/test.yaml -u https://example.com -silent
[FTL] Could not run nuclei: no valid templates were found

❯ ./nuclei -w workflow/basic.yaml -u https://example.com -silent

```

**Comment by dogancanbakir** [comment]
@ShubhamRasal Fixed failing template: https://github.com/projectdiscovery/nuclei-templates/pull/7060

**Review by ehsandeep — CHANGES_REQUESTED** [review]
@dogancanbakir @ShubhamRasal We can mark this condition as `SyntaxErrorStats` with `-validate` option to ensure we are not adding new templates in the template project and as `SyntaxWarningStats`  for the default run not to break/stop loading existing templates.

File: v2/pkg/parsers/parser.go [hunk 1]
```diff
@@ -68,10 +80,11 @@ func isTemplateInfoMetadataMatch(tagFilter *filter.TagFilter, template *template
 	return match, err
 }
 
-func validateTemplateFields(template *templates.Template) error {
+func validateTemplateFields(template *templates.Template) (err error, warning error) {
```
Comment (Mzack9999) [inline v2/pkg/parsers/parser.go:83]: What about splitting the function in `validateTemplateMandatoryFields` and `validateTemplateOptionalFields`?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
proposing function split

File: v2/pkg/parsers/parser.go (hunk 1 above)
Comment (ShubhamRasal) [reply v2/pkg/parsers/parser.go:83]: sounds nice... will update the functionality to split it

**Comment by Mzack9999** [comment]
@ShubhamRasal could you please complete the following task:
- Mention in the docs that optional severity policy is deprecated (optional during runtime execution but mandatory presence during template validation)

Thanks!

**Comment by ShubhamRasal** [comment]
> @ShubhamRasal could you please complete the following task:
> 
> * Mention in the docs that optional severity policy is deprecated (optional during runtime execution but mandatory presence during template validation)
> 
> Thanks!

https://github.com/projectdiscovery/nuclei-docs/pull/146

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**